### DISCUSSION TITLE
在一个ArrayField下array map生成一组Field,数组改变重新渲染之后，UI上只显示当前，数据上却保留了所有创建过的Field,该如何处理？

### DISCUSSION THREAD

**Discussion by specialCoder** [alibaba/formily#1667] [Q&A 2.x] [answered]
<ArrayField
        name="numbers"
      >
      {
        (field) => {
           return (field.value || []).map(({ week = '-', day = '-'},i) => {
              return <Field
                key={i}
                name={`W${week}D${day}`}
                title={`W${week}D${day}`}
                decorator={[FormItem]}
                component={[NumberPicker,{
                  precision:0,
                }]}
              />
            })
        }
      }
      </ArrayField>

**Comment by janryWang** [comment, marked as answer]
看文档，受控表单
